docs/spec/distribution/transactions.md

```diff
@@ -1,399 +1,208 @@
+    ValInfo = GetValidatorDistInfo(delegation.ValidatorAddr)
+    validator = GetValidator(delegation.ValidatorAddr)
 
-#### Delegation's entitlement to Global.Pool
+    // withdraw self-delegation
+    withdraw = GetDelegatorAllWithdraws(validator.OperatorAddr, height)
 
-For delegations (including validator's self-delegation) all fees from fee pool
-are subject to commission rate from the owner of the validator. The global
-shares should be taken as true number of global bonded shares. The recipients
-shares should be taken as the bonded tokens less the validator's commission.
+    // withdrawal validator commission rewards
+    global, commission = valInfo.WithdrawCommission(global, valInfo, height, pool.BondedTokens, 
+               validator.Tokens, validator.Commission)
+    withdraw += commission
+    SetGlobal(global) 
 
+    AddCoins(withdrawAddr, withdraw.TruncateDecimal())
 ```
-type DelegationFromGlobalPool struct {
-    DelegationShares              sdk.Dec
-    ValidatorCommission           sdk.Dec
-    ValidatorBondedTokens         sdk.Dec
-    ValidatorDelegatorShareExRate sdk.Dec
-    PoolBondedTokens              sdk.Dec
-    Global                        Global
-    ValDistr                      ValidatorDistribution
-    DelDistr                      DelegatorDistribution
-}
-
-func (d DelegationFromGlobalPool) DistributorTokens() DecCoins
-    return d.Global.Pool
-
-func (d DelegationFromGlobalPool) DistributorCumulativeTokens() DecCoins
-    return d.Global.EverReceivedPool
-
-func (d DelegationFromGlobalPool) DistributorPrevReceivedTokens() DecCoins
-    return d.Global.PrevReceivedPool
     
-func (d DelegationFromGlobalPool) DistributorShares() sdk.Dec
-    return d.PoolBondedTokens
+## Common calculations 
 
-func (d DelegationFromGlobalPool) DistributorPrevShares() sdk.Dec
-    return d.Global.PrevBondedTokens
+### Update total validator accum
 
-func (d DelegationFromGlobalPool) RecipientShares() sdk.Dec
-    return d.DelegationShares * d.ValidatorDelegatorShareExRate() * 
-           d.ValidatorBondedTokens() * (1 - d.ValidatorCommission)
+The total amount of validator accum must be calculated in order to determine
+the amount of pool tokens which a validator is entitled to at a particular
+block. The accum is always additive to the existing accum. This term is to be
+updated each time rewards are withdrawn from the system. 
 
-func (d DelegationFromGlobalPool) RecipientPrevShares() sdk.Dec
-    return d.DelDistr.PrevTokens
-
-func (d DelegationFromGlobalPool) RecipientAdjustment() sdk.Dec
-    return d.DelDistr.Adjustment
-
-func (d DelegationFromGlobalPool) ModifyAdjustments(withdrawal sdk.Dec)
-    d.ValDistr.Adjustment += withdrawal
-    d.DelDistr.Adjustment += withdrawal
-    d.global.Adjustment += withdrawal
-    SetValidatorDistribution(d.ValDistr)
-    SetDelegatorDistribution(d.DelDistr)
-    SetGlobal(d.Global)
+``` 
+func (g Global) UpdateTotalValAccum(height int64, totalBondedTokens Dec) 
```

Comment:
just an FYI that these receivers should be pointer receivers since they mutate self...

Reply:
yeah - to stick with the design pattern used elsewhere in the SDK, rather than using pointers, to address this comment, I've added returns for the modified object for each of these functions

Reply:
Are we strongly opposed to pointers? Pointers are definitely more efficient, and this is cost that can't really charge through gas.

Reply:
From my estimates, copying the struct means copying about 150 bytes, so I guess its not that much of a performance saving. (300 bytes total if you include the return value) I still would prefer pointers though.

Reply:
If there are considerably-large efficiency gains which will affect performance then it seems like a good idea. However it doesn't sound that way from your final comment.  In terms of code organization I AM strongly for non-pointers specifically because they enforce a code pattern which makes it very obvious where things are being written.  In the past there has been considerable development time wasted just deciphering what certain spaghetti code is doing because object writes are hidden several tiers down with no clear way to determine where things are being written.  For this reason - even a moderate-small efficiency boost doesn't justify using pointers throughout as it will make the code more prone to bugs. - I think there may be a situation in the future where we can use some special forms of linting or other tooling to aid code comprehension - after this is integrated it may be an appropriate time to switch to more pointer based code. THIS SAID under specific situations in the SDK pointers should of course still be used - we just need to make sure that we're hella explicit to make write locations obvious in the caller functions (and nested caller functions)

Reply:
Sounds good to me, thank you for the explanation!